src/main/scala/model/world/Generation.scala

```diff
@@ -54,18 +60,33 @@ object GenerationsUtils {
 
     /** @return the phase of the generation */
     def phase: Double
-  }
 
-  case class StartPhase(override val generationNumber: Int, override val phase: Double = START_PHASE)
-      extends GenerationPhase
-
-  case class WolvesPhase(override val generationNumber: Int, override val phase: Double = WOLVES_PHASE)
-      extends GenerationPhase
-
-  case class FoodPhase(override val generationNumber: Int, override val phase: Double = FOOD_PHASE)
-      extends GenerationPhase
+    /** @return after how many milliseconds from the start of generation there is the phase istant */
+    def instant: Double
+  }
 
-  case class HighTemperaturePhase(override val generationNumber: Int, override val phase: Double = TEMPERATURE_PHASE)
-      extends GenerationPhase
+  case class ReproductionPhase(
+      override val generationNumber: Int,
+      override val phase: Double = REPRODUCTION_PHASE,
+      override val instant: Double = GENERATION_END
+  ) extends GenerationPhase
+
+  case class WolvesPhase(
+      override val generationNumber: Int,
+      override val phase: Double = WOLVES_PHASE,
+      override val instant: Double = WOLVES_INSTANT
+  ) extends GenerationPhase
+
+  case class FoodPhase(
+      override val generationNumber: Int,
+      override val phase: Double = FOOD_PHASE,
+      override val instant: Double = FOOD_INSTANT
+  ) extends GenerationPhase
+
+  case class HighTemperaturePhase(
+      override val generationNumber: Int,
+      override val phase: Double = TEMPERATURE_PHASE,
+      override val instant: Double = TEMP_INSTANT
+  ) extends GenerationPhase
 
 }
```

Comment:
potremo aggiungere anche un'altra variabile che rappresenta il nome della generation phase o un metodo toString per ogni fase in modo tale che possa essere sfruttato anche da parte dell'orologio come stampa della fase corrente.

Reply:
Ci sta, questa parte ha sicuramente bisogno di un refactoring che direi che farò nel prossimo sprint